src/cpp/client.cpp

```diff
@@ -572,6 +572,13 @@ bool Client::tensor_exists(const std::string& name)
     return this->_redis_server->key_exists(g_key);
 }
 
+bool Client::dataset_exists(const std::string& name)
+{
+    // Same implementation as for tensors; the next line is NOT a type
+    std::string g_key = this->_build_tensor_key(name, true);
```

Comment:
We should probably use the ``_build_dataset_ack_key`` if we stick with the the current paradigm (see comment below for potential discussion).